tensorflow_addons/optimizers/tests/gradient_accumulator_test.py

```diff
@@ -198,3 +199,41 @@ def test_model_mixed_precision():
     opt = GradientAccumulator("sgd", trainable_variables=model.trainable_variables)
     model.compile(opt, loss="mse")
     model.fit(x, y, epochs=3)
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.needs_gpu
+def test_embedding():
+    def _get_dataset(vocab_size: int, batch_size: int = 10):
+        def _generator_fn():
+            size = np.random.randint(5, 10)
+            x = np.random.randint(low=0, high=vocab_size, size=size)
+            y = np.asarray([np.random.rand()])
+            yield x, y
+
+        return tf.data.Dataset.from_generator(
+            generator=_generator_fn,
+            output_types=(tf.int32, tf.float32),
+            output_shapes=((None,), (1,)),
+        ).padded_batch(batch_size)
+
+    strategy = tf.distribute.MirroredStrategy(test_utils.gpus_for_testing())
+
+    vocab_size = 10
+
+    with strategy.scope():
+        inputs = layers.Input(shape=(None,), dtype=tf.int32)
+        x = inputs
+        x = layers.Embedding(input_dim=vocab_size, output_dim=8)(x)
+        x = layers.Dense(1)(x)
+        model = tf.keras.models.Model(inputs=inputs, outputs=x)
```

Comment:
@fsx950223 If I use an `LSTM` layer here, it will crash. So this here is not working - probably the same issue we had previsously.

```python
    with strategy.scope():
        inputs = layers.Input(shape=(None,), dtype=tf.int32)
        x = inputs
        x = layers.Embedding(input_dim=vocab_size, output_dim=8)(x)
        # x = layers.LSTM(4)(x)  # <-- Not working
        x = layers.Dense(1)(x)
        model = tf.keras.models.Model(inputs=inputs, outputs=x)
```

Reply:
Just realized that you already told me that: https://github.com/tensorflow/addons/pull/2525#issuecomment-883102229 :)

Reply:
Okay, but there seems to be an issue with the LSTM layer. Running a training on multiple GPUs now gives me this `CancelledError`:

```
tensorflow.python.framework.errors_impl.CancelledError:  [_Derived_]RecvAsync is cancelled.
	 [[{{node GroupCrossDeviceControlEdges_0/GroupCrossDeviceControlEdges_0/cond/then/_0/cond/cond/branch_executed/_271/_133}}]]
	 [[cond_1/then/_12/cond_1/replica_1/Adam/ReadVariableOp_3/_92]] [Op:__inference_train_function_6953]
```

Reply:
Just test LSTM in a train script.

Reply:
Yeah, I added a test. For some reason it is crashing in the test but I am able to train my ASR model with it O_o

Reply:
Yeah, debug your model with ```tf.config.run_functions_eagerly(True)```